src/ProjectNeon/Assets/Features/Effects/Timed.cs

```diff
@@ -0,0 +1,37 @@
+﻿
+/**
+ * Wraps the execution of an effect so it is only executed for a certain amount of turns.
+ */
+public class Timed : ITemporalState, Effect
+{
+    
+    private int _remainingDuration;
+
+    private Effect _effect;
+
+    public IStats Stats { get; } = new StatAddends();
+    public bool IsDebuff => true;
```

Comment:
This should be passed in the `ctor`. Some effects are buffs, others are debuffs. Being Guarded by intercept is obviously a buff, and so we don't want it removed by a `Cleanse`